charts/cass-operator-chart/templates/deployment.yaml

```diff
@@ -13,6 +13,10 @@ spec:
         name: cass-operator
     spec:
       serviceAccountName: {{ .Values.serviceAccountName }}
+    {{- if .Values.imagePullSecret }}
+      imagePullSecrets:
+      - name: {{ .Values.imagePullSecret }}
+    {{- end }}
```

Comment:
So my feeling is that we should also put these image pull secrets in the PodSpec for the statefulsets in case someone wants to use a github package for the config-builder or maybe the management api (though we don't put that on GH packages presently). Note that the statefulsets do not share a service account with the operator, so we would have to have the operator add the pull secrets when creating the statefulsets. Alternatively, we could just make this the user's problem and have them add the image pull secrets to the CassandraDacenter resource themselves (or associate an appropriate service account).

Reply:
so this is just the cass-operator chart... I don't think we should propagate any secrets from the operator deployment itself into the statefulsets - the user put it here so they can put it in a service account that they put in their CassDC, or we can add `imagePullSecrets` to the CassDC spec

Reply:
Also, it might be nice to have the user just provide the registry name, username, and password and have the helm chart just make the secret for them. I did something similar for: https://github.com/datastax/cass-operator/pull/228